### ISSUE TITLE
Zero Search Results on Catalogsearch while Autocomplete Works

### ISSUE THREAD

**Issue by KholdenSP** [Smile-SA/elasticsuite#1586]
We run a multi-store install of Magento 2. All of our other sites run the search just fine. However, this particular URL shows zero results when you hit the catalogsearch page. Looking for some reasons as to why that would occur. Autocomplete works, so there is an index of some kind. 

### Preconditions
Magento Version : 2.3.1
ElasticSuite Version : 2.8
Environment : Production

### Steps to reproduce
1. Go to icelandicdesign.com
2. execute search for Sweaters

### Expected result
1. Autocomplete Shows
2. Search results show results

### Actual result
1. Autocomplete Shows
2. No Search results

![image](https://user-images.githubusercontent.com/53538134/67026478-ab4d5280-f0bc-11e9-96cb-cc6300b931a8.png)

![image](https://user-images.githubusercontent.com/53538134/67026507-b86a4180-f0bc-11e9-8382-78de312ea418.png)

Other sites on the same install using elasticsuite
bearpaw.com
aventuraclothing.com

Thanks,
Ken

**Comment by rbayet** [comment]
Hello @KholdenSP,

Considering your feature request #1587, I would have been tempted to point a problem to a third party extension/module messing with the `catalogsearch/result` page, but it should also impact your other sites as well.
Could you provide us anyway with a list of third party extension ?

Can you clarify that they are not affected at all ?
Can you clarify if the situation on icelandicdesign.com is systematic / for every search query or just for some search queries ?

If your problem is only encountered on icelandicdesign.com, then I would be tempted to point at something wrong in any custom design that you are using which would mess the layout.

If you can reproduce this behavior on a local or staging environment, I would suggest enabling Elasticsuite debug mode to check in var/log/system.log to see if indeed search requests are sent to Elasticsearch when hitting the fulltext search page.

Path is "Stores > Configuration > Elasticsuite > Base Settings > Elasticsearch Client"
![elasticsuite_debug_mode](https://user-images.githubusercontent.com/2353374/67157719-40a44d00-f330-11e9-91ba-70b1979e03d3.png)

Search requests logged will look like this :
```
[2019-10-20 09:35:56] report.INFO: curl -XPOST 'http://127.0.0.1:9200/magento2_default_catalog_product/product/_search?pretty=true' -d '{"size":20,"sort":[{"_score":{"order":"desc"}},{"entity_id":{"order":"desc","missing":"_first","unmapped_type":"keyword"}}],"from":0,"query":{"constant_score":{"filter":{"bool":{"must":[{"bool":{"must":[],"must_not":[],"should":[{"nested":{"path":"category","score_mode":"none","query":{"bool":{"must":[{"bool":{"must_not":[{"term":{"category.is_virtual":{"value":true,"boost":1}}}],"boost":1}},{"terms":{"category.category_id":["17"],"boost":1}}],"must_not":[],"should":[],"boost":1}},"boost":1}}],"minimum_should_match":1,"boost":1}},{"terms":{"visibility":[2,4],"boost":1}},{"terms":{"visibility":[2,4],"boost":1}}],"must_not":[],"should":[],"boost":1}},"boost":1}},"aggregations":{"attribute_set_id":{"terms":{"field":"attribute_set_id","size":100000,"order":{"_count":"desc"}}},"price.price":{"nested":{"path":"price"},"aggregations":{"price.price":{"filter":{"terms":{"price.customer_group_id":[0],"boost":1}},"aggregations":{"price.price":{"histogram":{"field":"price.price","interval":1,"min_doc_count":1}}}}}},"option_text_manufacturer":{"terms":{"field":"option_text_manufacturer.untouched","size":10,"order":{"_count":"desc"}}},"option_text_color":{"terms":{"field":"option_text_color","size":10,"order":{"_count":"desc"}}},"categories":{"filters":{"filters":[]}},"option_text_activity":{"terms":{"field":"option_text_activity","size":10,"order":{"_count":"desc"}}},"option_text_style_bags":{"terms":{"field":"option_text_style_bags","size":10,"order":{"_count":"desc"}}},"option_text_material":{"terms":{"field":"option_text_material","size":10,"order":{"_count":"desc"}}},"option_text_strap_bags":{"terms":{"field":"option_text_strap_bags","size":10,"order":{"_count":"desc"}}},"option_text_features_bags":{"terms":{"field":"option_text_features_bags","size":10,"order":{"_count":"desc"}}},"option_text_gender":{"terms":{"field":"option_text_gender","size":10,"order":{"_count":"desc"}}},"option_text_category_gear":{"terms":{"field":"option_text_category_gear","size":10,"order":{"_count":"desc"}}},"option_text_size":{"terms":{"field":"option_text_size","size":10,"order":{"_count":"desc"}}},"option_text_eco_collection":{"terms":{"field":"option_text_eco_collection","size":10,"order":{"_count":"desc"}}},"option_text_performance_fabric":{"terms":{"field":"option_text_performance_fabric","size":10,"order":{"_count":"desc"}}},"option_text_erin_recommends":{"terms":{"field":"option_text_erin_recommends","size":10,"order":{"_count":"desc"}}},"option_text_new":{"terms":{"field":"option_text_new","size":10,"order":{"_count":"desc"}}},"option_text_sale":{"terms":{"field":"option_text_sale","size":10,"order":{"_count":"desc"}}},"option_text_format":{"terms":{"field":"option_text_format","size":10,"order":{"_count":"desc"}}},"option_text_purpose":{"terms":{"field":"option_text_purpose","size":10,"order":{"_count":"desc"}}},"option_text_style_bottom":{"terms":{"field":"option_text_style_bottom","size":10,"order":{"_count":"desc"}}},"option_text_style_general":{"terms":{"field":"option_text_style_general","size":10,"order":{"_count":"desc"}}},"option_text_sleeve":{"terms":{"field":"option_text_sleeve","size":10,"order":{"_count":"desc"}}},"option_text_collar":{"terms":{"field":"option_text_collar","size":10,"order":{"_count":"desc"}}},"option_text_pattern":{"terms":{"field":"option_text_pattern","size":10,"order":{"_count":"desc"}}},"option_text_climate":{"terms":{"field":"option_text_climate","size":10,"order":{"_count":"desc"}}}}}' [] []
```

Regards,

**Comment by KholdenSP** [comment]
Sorry for slow reply. I'd be also tempted to blame a third party extension. Additionally, we've found that this is working correctly on staging, which tells me it's a setting or another extension causing the issue. Working with my developers to determine the root of the cause and will report back if it's something significant.

**Comment by KholdenSP** [comment]
Here is what the debug is showing...

icelanding search log:
{"image":"h.png","page":{"store_id":"5","type":{"identifier":"catalogsearch_result_index","label":"Quick Search Form"},"product_list":{"page_count":"1","product_count":"0","current_page":"1","sort_order":"relevance","sort_direction":"asc","display_mode":"grid"},"search":{"query":"sweater","is_spellchecked":false},"site":"icelandicdesign.com","url":"\/catalogsearch\/result\/","title":"Search results for: 'sweater' | Icelandic Design Official Site",**"referrer":{"domain":"icelandicdesign.com","page":"\/catalogsearch\/result\/"}**,"resolution":{"x":"1680","y":"949"}},"session":{"uid":"263c3601-15bf-55bc-4cf5-668753e5e82d","vid":"3697fcbd-f015-8339-076d-b5499c7820f5"}}

If I look at a good site on the same install,  I get this. 
bearpaw search log;
{"image":"h.png","page":{"store_id":"3","type":{"identifier":"catalogsearch_result_index","label":"Quick Search Form"},"product_list":{"page_count":"1","product_count":"1","current_page":"1","sort_order":"relevance","sort_direction":"asc","display_mode":"grid"},"search":{"query":"alyssa","is_spellchecked":false},"site":"bearpaw.com","url":"\/catalogsearch\/result\/","title":"Search results for: 'alyssa' | Bearpaw.com",**"referrer":{"domain":"bearpaw.com","page":"\/"},**"resolution":{"x":"1680","y":"949"}},"session":{"uid":"2fb9d93d-a6f6-d603-105b-9136c2bbeec3","vid":"403b08bc-f671-2ed1-f45b-b6cb239574f6"}}

The difference which I've highlighted, being:
"referrer":{"domain":"bearpaw.com","page":"\/"}," vs {"domain":"icelandicdesign.com","page":"\/catalogsearch\/result\/"},

Is it still a thought that a third party extension would cause that?

**Comment by rbayet** [comment]
@KholdenSP 

Those logs are not relevant, they are tracker related logs, not actual search requests/responses.
(And the differences between them are natural : those are "Site URL" dependent data).

I should have pointed it better, but the lines that interest you are those containing at the beginning an URL with "/product/_search?pretty.true", like : 
```
curl -XPOST 'http://127.0.0.1:9200/magento2_default_catalog_product/product/_search?pretty=true
```
The line matching this condition will be a category navigation/fulltext search request.

Regards,

**Comment by KholdenSP** [comment]
So the issue got solved in one of two ways. Both changes happened around the same time so I'm not sure which one solved it. 
1) We had an empty virtual category (nothing met the requirement). The devs moved it back to a non-virtual category and it seemed to work.
2) We had a layered navigation module enabled. For whatever reason, we had disabled this in staging, but it was enabled in production. I want to point to this one, but it's enabled on other stores without issue. Still disabling it anyway. 

Thanks for the help. Will forward the message regarding the logs.
